src/ui/opengl/castleinternalsettings.pas

```diff
@@ -261,25 +289,24 @@ procedure SettingsLoad(const Container: TUIContainer; const SettingsUrl: String)
         E.AttributeSingleDef('reference_height', DefaultUIReferenceHeight);
     end;
 
-    E := SettingsDoc.DocumentElement.Child('default_font', false);
-    if E <> nil then
+    NewFonts[ftRegular] := LoadFontSettings(SettingsDoc.DocumentElement.Child('default_font', false));
+    if NewFonts[ftRegular] = nil then
+      NewFonts[ftRegular] := LoadFontSettings(SettingsDoc.DocumentElement.Child('default_font_regular', false));
+    NewFonts[ftBold] := LoadFontSettings(SettingsDoc.DocumentElement.Child('default_font_bold', false));
+    NewFonts[ftItalic] := LoadFontSettings(SettingsDoc.DocumentElement.Child('default_font_italic', false));
+    NewFonts[ftBoldItalic] := LoadFontSettings(SettingsDoc.DocumentElement.Child('default_font_bolditalic', false));
+    if (NewFonts[ftBold] = nil) and (NewFonts[ftItalic] = nil) and
+      (NewFonts[ftBoldItalic] = nil) then
+      NewDefaultFont := NewFonts[ftRegular]
+    else
     begin
-      DefaultFontUrl := E.AttributeURL('url', SettingsUrl);
-      DefaultFontSize := E.AttributeCardinalDef('size', 20);
-      DefaultFontAntiAliased := E.AttributeBooleanDef('anti_aliased', true);
-
-      if E.AttributeString('sizes_at_load', AllSizesAtLoadStr) then
-      begin
-        AllSizesAtLoad := ParseIntegerList(AllSizesAtLoadStr);
-        NewDefaultFont := TCustomizedFont.Create(Container);
-        TCustomizedFont(NewDefaultFont).Load(DefaultFontUrl, AllSizesAtLoad, DefaultFontAntiAliased);
-      end else
-      begin
-        DefaultFontLoadSize := E.AttributeCardinalDef('size_at_load', DefaultFontSize);
-        NewDefaultFont := TTextureFont.Create(Container);
-        TTextureFont(NewDefaultFont).Load(DefaultFontUrl, DefaultFontLoadSize, DefaultFontAntiAliased);
-      end;
-      NewDefaultFont.Size := DefaultFontSize;
+      NewFontFamily := TFontFamily.Create(Container);
+      NewFontFamily.Name := 'FontVariants';
```

Comment:
Assigning the `NewFontFamily.Name` here is risky, as user may create his/her own TComponent with such name, and set as `Owner` the `Container`. And all components with the same owner must have unique names.

Sure, there's only a small chance that someone will use exactly this name, but in general we want to leave all `Name` (that can affect user) to be configurable by developer.

Hm, or we can clearly mark the `Name` as something that will never be used by developer. Instead of `FontVariants`, I recommend `CastleInternalDefaultFont`. There's no way someone will use it by accident (without knowingly causing clash :) ).